pkg/operator/apiserver/controller/workload/workload.go

```diff
@@ -353,6 +375,33 @@ func isUpdatingTooLong(operatorStatus *operatorv1.OperatorStatus, progressingCon
 	return progressing != nil && progressing.Status == operatorv1.ConditionTrue && time.Now().After(progressing.LastTransitionTime.Add(15*time.Minute)), nil
 }
 
+// hasDeploymentProgressed returns true if the deployment reports NewReplicaSetAvailable
+// via the DeploymentProgressing condition
+func hasDeploymentProgressed(status appsv1.DeploymentStatus) bool {
+	for i := range status.Conditions {
+		c := status.Conditions[i]
+		if c.Type == appsv1.DeploymentProgressing {
```

Comment:
Would it make sense taking `workloadIsBeingUpdated := workload.Status.UpdatedReplicas < desiredReplicas` into account in addition to this? or `NewReplicaSetAvailable` is already covering?

Reply:
that is already  covered by the condition
- https://github.com/kubernetes/kubernetes/blob/5ffb0528dd0ca2faa9c8c4c360216268926ee0ee/pkg/controller/deployment/progress.go#L60
- https://github.com/kubernetes/kubernetes/blob/5ffb0528dd0ca2faa9c8c4c360216268926ee0ee/pkg/controller/deployment/util/deployment_util.go#L708

There can be a short amount of time when this condition is stale, before the deployment controller observes the new revision - so I made the `workloadIsBeingUpdated` variable depend also on the `workloadAtHighestGeneration` and added a test case for that.